### ISSUE TITLE
Upgrading xonsh obliterates installed xontribs

### ISSUE THREAD

**Issue by astronouth7303** [xonsh/xonsh#2411]
I suspect this is a problem with pip?

When I upgraded my xonsh (`xip install --upgrade xonsh`), it removed `xontrib/avox.py` and `xontrib/z.py` without actually uninstalling them.

**Comment by scopatz** [comment]
Uggh, I think pip still has some issues with implicit namespace packages. Probably worth reporting.

**Comment by astronouth7303** [comment]
I feel like I need a reduced test case to test this, but ugggg.

**Comment by scopatz** [comment]
Maybe for setuptools we need to use the `namespace_packages` kwargs http://setuptools.readthedocs.io/en/latest/setuptools.html#namespace-packages

**Comment by scopatz** [comment]
Super annoying =/

**Comment by astronouth7303** [comment]
Oh, that's new? But the bit about `__init__.py` is a Py2 thing.

**Comment by scopatz** [comment]
Yeah, so I'm guessing that we can ignore the `__init__.py` stuff.

**Comment by astronouth7303** [comment]
* [PyPA's documentation on the matter](https://packaging.python.org/guides/packaging-namespace-packages/)
* [A related bug](https://github.com/pypa/python-packaging-user-guide/issues/265)
* [The tests](https://github.com/pypa/sample-namespace-packages)

**Comment by gforsyth** [comment]
We should probably add a note to the `xontrib` docs (or in the warning that "you tried to load this but it isn't installed") that to re-enable them you have to force reinstall.
The not removing them is super annoying, since a `xpip install --upgrade xonda` will just say it's already there.  You have to add a `--force` to get it to properly update to the next install.

**Comment by scopatz** [comment]
:+1: to that @gforsyth!
